lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -192,19 +192,24 @@ from B<_generate_jobs()>. Returns a list of job ids from the jobs that were succ
 and a list of failure reason for the jobs that could not be scheduled. Internal function, not
 exported - but called by B<create()>.
 
+=item $guard
+
+Expects an C<OpenQA::Task::SignalGuard> object. If database transcation fails during the scheduling,
+this ensures that the minion job will be restarted.
```

Comment:
First a spelling/grammar nitpick:

```suggestion
Expects an C<OpenQA::Task::SignalGuard> object. If the database transaction fails during the scheduling,
this ensures that the Minion job will be restarted.
```

But this explanation is also completely the opposite of what passing the `$guard` will achieve here. It achieves the following:

```suggestion
Expects an C<OpenQA::Task::SignalGuard> object. If the related Minion job is aborting after the database
transaction has completed, a restart of the Minion job is prevented so jobs aren't created twice.
```